src/components/shared/Map/MapPointer.tsx

```diff
@@ -0,0 +1,21 @@
+import L from 'leaflet';
+import { Circle, Marker } from 'react-leaflet';
+
+interface MapPointerProps {
+  center: [number, number];
+  radius: number;
+  text: string;
+  color: string;
+}
+
+function MapPointer({ center, radius, text, color }: MapPointerProps) {
+  const text1 = L.divIcon({ className: 'text-2xl font-bold', html: text });
+
+  return (
+    <Circle center={center} radius={radius} fillColor={color} color={color}>
+      <Marker position={center} icon={text1} />
+    </Circle>
```

Comment:
you forgot to add a click handler to open a popup.
define 
`const circleRef = useRef<L.Circle>(null);`

add `sensorData` to MapPointerProps:
`
interface MapPointerProps {
  center: [number, number];
  radius: number;
  text: string;
  color: string;
  sensorData: SensorData;
}
`

then bind to the Circle ref:
`
<Circle
    ...
    ref={circleRef}>
...
</Circle>
`

then add event handler on Marker component:
`
 <Marker
        ...
        eventHandlers={{ click: () => circleRef.current?.openPopup() }}
      />
`

finally add popup as a sibling to the `<Marker/>` inside `<Circle>`
`
<Popup>
        <PopupComponent sensorData={sensorData} />
      </Popup>
`